packages/react/src/components/RadioButtonGroup/next/RadioButtonGroup.js

```diff
@@ -0,0 +1,138 @@
+import PropTypes from 'prop-types';
+import React, { useState, useEffect } from 'react';
+import classNames from 'classnames';
+import { Legend } from '../../Text';
+import { usePrefix } from '../../../internal/usePrefix';
+
+const RadioButtonGroup = React.forwardRef(function RadioButtonGroup(
+  {
+    children,
+    className,
+    defaultSelected,
+    disabled,
+    labelPosition = 'right',
+    legendText,
+    name,
+    onChange = () => {},
+    orientation = 'horizontal',
+    valueSelected,
+  },
+  ref
+) {
+  const prefix = usePrefix();
+
+  const [selected, setSelected] = useState(
+    valueSelected !== 'undefined' ? valueSelected : defaultSelected
+  );
+  const [prevValueSelected, setPrevValueSelected] = useState(selected);
+
+  /**
+   * prop + state alignment - getDerivedStateFromProps
+   * only update if selected prop changes
+   */
+  useEffect(() => {
+    if (prevValueSelected !== valueSelected) {
+      setSelected(
+        valueSelected !== 'undefined' ? valueSelected : defaultSelected
+      );
+      setPrevValueSelected(valueSelected);
+    }
+  }, [valueSelected]); //eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
We can use an `if` here instead of wrapping in an effect 👍 

```suggestion
    if (prevValueSelected !== valueSelected) {
      setSelected(
        valueSelected !== 'undefined' ? valueSelected : defaultSelected
      );
      setPrevValueSelected(valueSelected);
    }
```